### PR TITLE
feat(kcollapse): component reskin [KHCP-8976]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

[Figma](https://www.figma.com/file/Yze0SWXl5nKjR0rFdilljK/Components?type=design&node-id=2020%3A11254&mode=dev&t=z4m6LiZVAeHHwxSH-1)
[Jira](https://konghq.atlassian.net/browse/KHCP-8976)

`KCollapse` component reskin

Links:

- [Kongponents sandbox](https://deploy-preview-2125--kongponents-sandbox.netlify.app/#/collapse)
- [Kongponents docs preview](https://deploy-preview-2125--kongponents.netlify.app/components/collapse.html)

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

**Review by mptap — APPROVED** [review]
Functionally verified in sandbox 🎉 

Screenshot 2024-04-10 at 10 25 08 AM

File: src/components/KCollapse/KCollapse.vue [hunk 1]
```diff
@@ -1,81 +1,79 @@
...
-      <slot />
-    </div>
+    <Transition name="kongponents-fade-transition">
+      <div
+        v-show="!collapsedState"
+        :id="contentId"
+        class="collapse-hidden-content"
+        data-testid="collapse-hidden-content"
+      >
+        <slot />
```
Comment (adamdehaven) [inline src/components/KCollapse/KCollapse.vue:63]: nit, just so it's clear
```suggestion
        <slot name="default" />
```

File: docs/components/collapse.md [hunk 2]
```diff
@@ -1,257 +1,241 @@
...
+  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
+</KCollapse>
 ```
 
+### titleTag
+
+HTML element you want title to be rendered as. Defaults to `div`.
+
+Accepted values are:
+* `div`
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/components/collapse.md:36]: Done

File: src/components/KCollapse/KCollapse.vue [hunk 3]
```diff
@@ -1,81 +1,79 @@
...
+            :aria-controls="contentId"
+            :aria-expanded="!collapsedState"
+            :aria-label="triggerLabel ? undefined : 'Toggle content'"
+            class="collapse-trigger-content"
+            data-testid="collapse-trigger-content"
             role="button"
             @click.prevent.stop="toggleDisplay()"
           >
             <slot name="trigger-content">
+              <ChevronRightIcon
```
[the remark this answers is not in the document]
Comment (portikM) [reply src/components/KCollapse/KCollapse.vue:31]: Yes

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**